src/Map.php

```diff
@@ -3727,4 +3727,34 @@ protected function visit( iterable $entries, array &$result, int $level, ?\Closu
 			}
 		}
 	}
+
+
+	/**
+	 * Executes $callback if the Map is empty
+	 *
+	 * Examples: (Returns match or dies if doesn't exist)
+	 *  $page = $this->pages
+	 *          ->filter(fn(Page $page, $_) => $page->Id == $pageId)
+	 *          ->if(
+	 *              fn(Map &$Map) => $Map->isEmpty(),
+	 *              fn(Map &$_) => die("unknown page: " . $pageId))
+	 *          ->first();
+	 *
+	 * @param callable $condition Function with (Map) parameter and returns bool
+	 * @param callable $then Function with (Map) parameter and returns void
+	 * @param callable|null $else Optional function with (Map) parameter and returns void
+	 * @return self Same map for fluid interface
+	 */
+	public function if(callable $condition, callable $then, callable $else = null): self
```

Comment:
Can you please remove the first `callable` type hint so it can be also something that evaluates to a boolean?

Reply:
Resolved in 7b6ad457a2dbdf619d0e3d91ea67f009ff626a4d